planner/core/rule_predicate_simplification.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"context"
+
+	"github.com/pingcap/tidb/expression"
+	"github.com/pingcap/tidb/parser/ast"
+)
+
+// predicateSimplification consolidates different predcicates on a column and its equivalence classes.  Initial out is for
+// in-list and not equal list intersection.
+type predicateSimplification struct {
+}
+
+type predicateType = byte
+
+const (
+	inListPredicate   predicateType = 0x00
+	notEqualPredicate predicateType = 0x01
+	otherPredicate    predicateType = 0x02
```

Comment:
```suggestion
	inListPredicate   predicateType = iota
	notEqualPredicate
	otherPredicate 
```
In golang, this syntactic sugar should work

Reply:
Sounds good. Will be done in next round.